templateDataGenerator.go

```diff
@@ -157,6 +157,9 @@ func generateTemplateData(params Params, currentReplicas int, gitSource, gitOwne
 
 	data.Container.Readiness.IncludeOnContainer = !data.HasOpenrestySidecar || params.Container.ReadinessProbe.Port != params.Container.Port || params.Container.ReadinessProbe.Path != params.Sidecar.HealthCheckPath
 
+	// if container port is set to 443, we always use https named port
+	data.UseHTTPS = data.HasOpenrestySidecar || params.Container.Port == 443
```

Comment:
Using container port 443 is only allowed when there's no openresty sidecar, otherwise the use of that port will collide. So should be mutually exclusive in some way. Is that already the case?

Reply:
I am just looking to set `nginx.ingress.kubernetes.io/backend-protocol: "HTTPS"` without the need of the openresty sidecar. So you either have the sidecar or you have port 443 set.
By mutually exclusive you mean failing early by detecting the port?

Reply:
If the openresty sidecar is used it will try to listen on port 443, so the main application container can't use the same port. Hence setting container.port=443 should only be possible if no openresty sidecar is used. 

Without openresty the `svc` will already point to the main container port, see https://github.com/estafette/estafette-extension-gke/blob/master/templates/service.yaml#L45. Not sure any ingress will point to this `web` port though, so might only work for internal communication.

Reply:
That was my first idea. I just didn't want to add a new parameter. But I guess it makes sense.

Reply:
The problem of changing those lines on the template, is I would have to make some checks also in the deployment.yaml template. Which would make it even more complex.
I think we're better like it is now, just allowing one simple case. If this gets any traction we can improve it in the future.

Reply:
If you can add just the validation check to ensure openresty and container port 443 are not used at the same time I think we can postpone the scheme parameter to a later PR.